OpenSim/Common/Property.h

```diff
@@ -938,6 +952,88 @@ readSimplePropertyFromStream(std::istream& in)
        return SimTK::readUnformatted(in, values);
 }
 
+// We provide specializations for all types that contain floats to manage
+// the number of digits displayed. We rely on the default setting of how
+// `stringstream`s handle floats, where `precision` is the number of digits
+// printed, and the `stream` automatically determines whether float or
+// exponential notation is used.
+template <> inline void SimpleProperty<double>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << values[i];
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vec3>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(" << values[i][0] << " ";
+        o << values[i][1] << " ";
+        o << values[i][2] << ")";
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vec6>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(" << values[i][0] << " ";
+        o << values[i][1] << " ";
+        o << values[i][2] << " ";
+        o << values[i][3] << " ";
+        o << values[i][4] << " ";
+        o << values[i][5] << ")";
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Vector>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
+    for (int i = 0; i < values.size(); ++i) {
+        if (i != 0) o << " ";
+        o << "(";
+        for (int j = 0; j < values[i].size(); ++i) {
+            if (j != 0) o << " ";
+            o << values[i][j];
+        }
+        o << ")";
+    }
+}
+
+template <> inline void SimpleProperty<SimTK::Transform>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    // Convert array of Transform objects to an array of Vec6 objects.
+    SimTK::Array_<SimTK::Vec6> rotTrans;
+    for (int i = 0; i<values.size(); ++i) {
+        SimTK::Vec6 X6;
+        SimTK::Vec3& angles = X6.updSubVec<3>(0);
+        SimTK::Vec3& pos = X6.updSubVec<3>(3);
+        angles = values[i].R().convertRotationToBodyFixedXYZ();
+        pos = values[i].p();
+        rotTrans.push_back(X6);
+    }
```

Comment:
Consider creating a helper function `convertTransformToVec6()` (probably operating on Arrays), to avoid duplicating this code.